src/main/java/org/grouplens/grapht/Instantiators.java

```diff
@@ -140,13 +146,20 @@ public ProviderInstantiator(Instantiator prov) {
         @Override
         public Object instantiate() throws ConstructionException {
             Provider<?> provider = (Provider) providerInstantiator.instantiate();
+            LogContext mdcContextProvider = LogContext.create();
+            logger.trace("invoked provider",provider);
```

Comment:
English tense and logging placeholder: should be `logger.trace("invoking provider {}", provider);`.